examples/knative/python/kn-py-vm-attr/handler.py

```diff
@@ -0,0 +1,98 @@
+from flask import Flask, request, jsonify
+from cloudevents.http import from_http
+import logging, json
+from vcenter import Session
+from datetime import date
+
+logging.basicConfig(level=logging.DEBUG,format='%(asctime)s %(levelname)s %(name)s %(threadName)s : %(message)s')
+
+app = Flask(__name__)
+@app.route('/', methods=['GET','POST'])
+def echo():
+    if request.method == 'GET':
+        sc = 200
+        msg = 'POST to this endpoint to apply custom attributes to a VM object within a cloudEvent message'
+        message = {
+            'status': sc,
+            'message': msg,
+        }
+        resp = jsonify(message)
+        resp.status_code = sc
+        return resp
+
+    if request.method == 'POST':
+        try:
+            event = from_http(request.headers, request.get_data(),None)
+
+            data = event.data
+            # hack to handle non JSON payload, e.g. xml
```

Comment:
This hack won't work for your example. You could check for the `https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#datacontenttype` attribute and reject any non-JSON payload with a 400 error message